Multicamera support

mlehtima opened the pull request:
Rename direction key in quirks to camera_id. Add property for camera direction.

Supersedes https://github.com/sailfishos/gst-droid/pull/56

mlehtima committed:
[gst-droid] Rename direction key in quirks to camera_id.

mlehtima committed:
[gst-droid] Add property for camera direction.

abranson commented on gst/droidcamsrc/gstdroidcamsrc.h:
```diff
@@ -48,7 +48,7 @@ G_BEGIN_DECLS
   (G_TYPE_CHECK_CLASS_TYPE((klass), GST_TYPE_DROIDCAMSRC))
 
-#define MAX_CAMERAS 2
+#define MAX_CAMERAS 255
```
Is this still needed? Would we ever not be able to support whatever android_media_camera_get_number_of_cameras declares?

abranson commented on gst/droidcamsrc/gstdroidcamsrc.c:
```diff
@@ -105,14 +105,15 @@ enum
 static guint droidcamsrc_signals[LAST_SIGNAL];
 
-#define DEFAULT_CAMERA_DEVICE          GST_DROIDCAMSRC_CAMERA_DEVICE_PRIMARY
+#define DEFAULT_CAMERA_DEVICE          0
```
It would have been useful in the past to have had this configurable. The Gemini without the extra camera module only had the front-facing camera 1. It might be simpler now to drop this completely, and choose the default as the lowest numbered camera that is present.

mlehtima committed:
[gst-droid] Initial multiple camera support. JB#53099

Dynamically fetch information for all cameras from droidmedia. Change camera device from enum to integer.

Co-authored-by: Michal Szczepaniak <m.szczepaniak.000@gmail.com>

adenexter commented on gst/droidcamsrc/gstdroidcamsrc.c:
```diff
@@ -482,29 +479,15 @@ gst_droidcamsrc_get_hw (GstDroidCamSrc * src)
-    return FALSE;
+    found = gst_droidcamsrc_fill_info (src, &src->info[x], x);
+    if (!found) {
+      GST_WARNING_OBJECT (src, "cannot find camera %d", x);
```
If there's an issue with an item at an index it should probably be removed from the set of cameras.

mkosola replied:
Yeah, we should fix this later. There is slightly small change that this will cause problem. I'll merge this now as qtmultimedia got already merged.
